### DISCUSSION TITLE
Transposing tuning

### DISCUSSION THREAD

**Discussion by circinusX1** [CoderLine/alphaTab#636] [Ideas]
I am trying to find where the guitar tuning is parsed in the alhatab.js Ver:1/3/0 to 
transpose the tuning like for example:    Eb Ab Gb Gb Bb Eb - > E A D G B E (standard)
I am trying this, and looks promising. It keeps the string/frets and adjusts the 
notes. Later on after each note has the fret/string I do some corrections if the tunes/string
have different offsets.

```

           ////////////    overwrite the tuning before readBars().... 
            if(this.settings.mco_tune && this.settings.mco_tune.length){
                for (let i = 0; i < this._score.tracks.length; i++) {
                    for (let s = 0; s < this._score.tracks[i].staves.length; s++) {
                        if(this._score.tracks[i].staves[s].stringTuning.isStandard==false){
                            this._score.tracks[i].staves[s].stringTuning.isStandard=true;
                            this._score.tracks[i].staves[s].stringTuning.name="";
                            this._score.tracks[i].staves[s].stringTuning.tunings_original=this._score.tracks[i].staves[s].stringTuning.tunings;
                            this._score.tracks[i].staves[s].stringTuning.tunings=this.settings.mco_tune;
                           
                        }
                    }
                }


```

If you have a better idea would be appreciated.

**Comment by Danielku15** [comment]
In general for stringed instruments there are two aspects on the final displayed notes: the string tuning of the related staff, but also the fret/string the individual notes are played. 

So it strongly depends on what you actually want as feature to users. If you want to offer a feature where you can change the tuning, but keep the original note height, you will have to adjust the string and fret of each individual note in the model to fit the new tuning. 

The code you're showing, will change the tuning, but keep the string/frets of the notes. This will change the display of the tuning annotation and the note height of the standard notation will adapt. 

Maybe you want to share some insights on the actual feature you're trying to build. This way I might be able to help better. 😉
